qiskit/pulse/instructions/directives.py

```diff
@@ -55,3 +55,80 @@ def channels(self) -> Tuple[chans.Channel]:
     def __eq__(self, other):
         """Verify two barriers are equivalent."""
         return isinstance(other, type(self)) and set(self.channels) == set(other.channels)
+
+
+class AreaBarrier(Directive):
+    """Pulse ``AreaBarrier`` directive.
+
+    This instruction is intended to be used internally within the pulse builder,
+    to naively convert :class:`.Schedule` into :class:`.ScheduleBlock`.
```

Comment:
If this is used for a `naive` conversion then what would a non-naive conversion look like?

Reply:
In the current mechanism I "give up" to infer the original alignment context. For example,
```python
with align_sequential():
    with align_left():
        play(cr45p, ControlChannel(1))
        play(cr45p_d0, DriveChannel(0))
   play(x, DriveChannel(1))
    with align_left():
        play(cr45m, ControlChannel(1))
        play(cr45m_d0, DriveChannel(0))
   play(x, DriveChannel(1))
```
this would be the context of the ECR schedule, but current mechanism will generate
```python
with align_left():
    play(cr45p, ControlChannel(1))
    play(cr45p_d0, DriveChannel(0))
    time_block(xxx, DriveChannel(1))
    play(x, DriveChannel(1))
    ...
```
Note that we cannot modify any pulse duration (otherwise you must also modify time block duration), but this should be okey because we are consuming "already scheduled" sequence. To recover original alignment context, we may need to generate DAG to find parallel context. Such logic may heavy and still has a rick of some edge case. So I just avoid doing over engineering and choose to say "this is naive version of alignment context recovery".

Reply:
Added bit more detail in https://github.com/Qiskit/qiskit-terra/pull/8949/commits/7ab2710f05605a1d70bbd3d5405243092d0340c1